mart/callbacks/progress_bar.py

```diff
@@ -4,19 +4,47 @@
 # SPDX-License-Identifier: BSD-3-Clause
 #
 
-import tqdm
+from typing import Any
+
+import pytorch_lightning as pl
 from pytorch_lightning.callbacks import TQDMProgressBar
+from pytorch_lightning.utilities.rank_zero import rank_zero_only
 
 __all__ = ["ProgressBar"]
 
 
 class ProgressBar(TQDMProgressBar):
     """Display progress bar of attack iterations with the gain value."""
 
+    def __init__(self, disable=False, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if disable:
+            self.disable()
+
+        # rank starts with 0
+        rank_id = rank_zero_only.rank
+        # Adversary progress bars start at position 1, because the main progress bar takes position 0.
+        self._process_position = rank_id + 1
```

Comment:
Don't set private variables. Instead set process_position via super().__init__().

Reply:
Resolved in [771e181](https://github.com/IntelLabs/MART/pull/153/commits/771e1810df21bd54aad46d5f3c7b3c967e2cc47a)